src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -31,32 +32,28 @@ function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
   }
 }
 
-function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
-  // WalletConnect relies on Buffer, so it must be polyfilled.
-  if (!('Buffer' in window)) {
-    window.Buffer = Buffer
-  }
-
-  // FIXME(kristiehuang): we don't know what the props.jsonRpcEndpoint chain is; assume mainnet for WC instantiation
-  let mainnetRpcUrl: string | undefined
+function getWalletConnectConnection(
+  useDefault: boolean,
+  jsonRpcEndpoint: string | JsonRpcProvider | { [chainId: number]: string[] },
+  defaultChainId?: number
+) {
+  let urlMap: { [chainId: number]: string[] }
   if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
-    mainnetRpcUrl = jsonRpcEndpoint.connection.url
+    urlMap = { [defaultChainId ?? SupportedChainId.MAINNET]: [jsonRpcEndpoint.connection.url] }
+  } else if (typeof jsonRpcEndpoint === 'string') {
+    // if integrator only provides a jsonRpcEndpoint string, user switching networks on their wallet will not work as expected
```

Comment:
imo we should throw in this case - we shouldn't allow devs to build experiences that are confusing for users

Reply:
Agreed, let's remove it. The reason this was in was for backwards compatibility but since (as Kristie reminded) this is going to be a breaking change anyway, it's fine.